### DISCUSSION TITLE
If I built a purpose built VM what OS would I be recommended to use?

### DISCUSSION THREAD

**Discussion by turtle2472** [jasonacox/Powerwall-Dashboard#251] [General]
Basically I'm looking to build this whole dashboard "environment" in a single purpose VM to run in an ESXi host I have.  This will make many things easier for me.  My general preference is CentOS but I'm open to all flavors of linux.     

To continue the thought, on a green field build, what installation option is going to result in the least customizing and easiest to maintain?

**Comment by jasonacox** [comment]
I'm running my test infrastructure with [Ubuntu](https://ubuntu.com/).  I tend to prefer the `apt` versus `yum` package manager, but technically any Linux installation that supports docker and docker-compose should work (and podman too). Some are even using NAS systems (Synology) to host the project containers.

**Reply by turtle2472** [reply]
Thanks for the input.  If it is what you test with I can consider it good enough.  That is the perk of building VMs, just tear them down if they don't work out like you thought they would. :D

**Comment by anderep** [comment]
I tried on clearOS and the firewall port stuff wouldn’t configure. Raspberry Pi is easiest.  Pi 4b, 8gB  only uses about 1.5 watts.
